### PR TITLE
Ar mock service

### PR DESCRIPTION
**Pull request by aracioppibroad**
Mock Services for vault-api.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/vault/services/VaultApiServiceHandlerProps.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  def getDescribeServiceHandler(requestContext: RequestContext, dmService: ActorRef ): Props
+}
+
+/**
+ * Extends the Service Provider and returns the real services.
+ */
+
+trait StandardVaultApiServiceHandlerProps extends VaultApiServiceHandlerProps{
+  def getDmClientService(requestContext: RequestContext)=
+    Props(new DmClientService(requestContext))
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/vault/services/VaultApiServiceHandlerProps.scala:20]: Use DmClientService.props() here.

http://doc.akka.io/docs/akka/2.2.3/scala/actors.html#Recommended_Practices

**Comment by jmthibault79** [comment]
`build.sbt` and `src/test/scala/org/broadinstitute/dsde/vault/services/uBAM/DescribeServiceSpec.scala` did not actually change in this PR.  For a cleaner git history, please revert them to their original versions in origin/develop.

**Comment by rushtong** [comment]
:+1: In general, I think this approach works.

**Comment by davidangb** [comment]
I'll throw a total wrench in here ... what about using something like MockServer (http://www.mock-server.com/) or WireMock (http://wiremock.org/) to create a fake vault-datamanagement and fake BOSS for vault-api to talk to? This would leave all of the vault-api code alone, and we'd just have to point it at a different url for its dependent services.

Apparently there are services that will do this for us, too: https://www.mockable.io, https://getsandbox.com/. 

Kind of an ugly option: use swagger-codegen to generate a server for us. Stand up that server and use it for unit tests.

I really do like the solution you came up with, but it seems more applicable for mocking out logic contained within Actors ... and we need to be testing those Actors!

**Comment by tedsharpe** [comment]
My strategy for mocking the objectStore in BOSS was to take advantage of the fact that it already is a web server!  I added a new dropwizard resource that implements FCS -- the Faux Cloud Storage system.  It is put into play only when necessary (i.e., typically just during unit testing), so it doesn't clutter up a real server.  Just a thought.

**Comment by rushtong** [comment]
I'm going to walk back my assessment of the proposed approach. If we used something like MockServer, we would have a _lot_ less impact on the existing test code. We would have a mock server for each external service (dm, boss) and override the config urls for unit tests. That's it.

I'm less a fan of the hosted solutions. They would require some small amount of coordination to make sure the required query responses are in place before testing. In the back and forth of updating/testing, that can be a slowdown. If the APIs are stable, I think they are a more attractive solution.

**Comment by dags** [comment]
Our first approach was to generate a fake server. We did it in the past 
using wsdl or wadl files and code generation. We ended up using a more 
"classical" approach, injecting mocked services. If the code is designed 
with this in mind, it may be the simplest way.

Mock servers is an interesting approach too, I'll take a look to know 
each one better.

Not always is acceptable to rely on external servers or code for 
testing: we may need additional servers to deploy mock servers on, 
additional deployment logic, the
re will be more point of failures, etc. As always, it is a cost - 
benefits issue.

Regards,

Diego A. Gil
Java Developer
_Belatrix Software_
Mail: dgil@belatrixsf.com mailto:dgil@belatrixsf.com
USA Phone +1 (617) 608-1413
AR Phone +54(261) 542-3033 Ext. 1189
Skype ID: diego.a.gil
http://www.belatrixsf.com

_WARNING OF CONFIDENTIALITY _ : The information contained and 
transmitted here is CONFIDENTIAL and it is for exclusive use of the 
addressee indicated above, and for his/her specific use. If you are not 
the addressee, we apologize for any inconvenience. It is hereby notified 
that it is prohibited to revise, retransmit or broadcast or any other 
type of use of the information contained herein by people who are not 
the original addressee. If you have received this information by 
mistake, please contact the sender and eliminate the information 
contained here from all computers.

_AVISO DE CONFIDENCIALIDAD _ : La información aquí contenida y 
transmitida es CONFIDENCIAL, para uso exclusivo del destinatario arriba 
indicado y para su utilización específica. Si usted no es el 
destinatario, sepa disculpar la molestia. Se le notifica por el presente 
que está prohibida su revisión, retransmisión, difusión, y/o cualquier 
otro tipo de uso de la información contenida por personas extrañas al 
destinatario original. Si Ud. Ha recibido por error esta información, 
por favor contacte al remitente y elimine la información aquí contenida 
de toda computadora donde resida.

**Pull request closed without merging**